screen2.0/src/app/applets/gene-expression/page.tsx

```diff
@@ -0,0 +1,790 @@
+    if (checkList.includes(option)) {
+      const index = checkList.indexOf(option, 0)
+      if (index > -1) {
+        checkList.splice(index, 1)
+      }
+    } else {
+      checkList.push(option)
+    }
+
+    return checkList
+  }
+
+  // gene descriptions
+  useEffect(() => {
+    const fetchData = async () => {
+      let f = await Promise.all(
+        options.map((gene) =>
+          fetch("https://clinicaltables.nlm.nih.gov/api/ncbi_genes/v3/search?authenticity_token=&terms=" + gene.toUpperCase())
+            .then((x) => x && x.json())
+            .then((x) => {
+              const matches = (x as QueryResponse)[3] && (x as QueryResponse)[3].filter((x) => x[3] === gene.toUpperCase())
+              return {
+                desc: matches && matches.length >= 1 ? matches[0][4] : "(no description available)",
+                name: gene,
+              }
+            })
+            .catch(() => {
+              return { desc: "(no description available)", name: gene }
+            })
+        )
+      )
+      setgeneDesc(f)
+    }
+
+    options && fetchData()
+  }, [options])
+
+  // gene list
+  const onSearchChange = async (value: string) => {
+    setOptions([])
+    const response = await fetch("https://ga.staging.wenglab.org/graphql", {
+      method: "POST",
+      body: JSON.stringify({
+        query: GENE_AUTOCOMPLETE_QUERY,
+        variables: {
+          assembly: "GRCh38",
+          name_prefix: value,
+          limit: 100,
+        },
+      }),
+      headers: { "Content-Type": "application/json" },
+    })
+    const genesSuggestion = (await response.json()).data?.gene
+    if (genesSuggestion && genesSuggestion.length > 0) {
+      const r = genesSuggestion.map((g) => g.name)
+      const g = genesSuggestion.map((g) => {
+        return {
+          chrom: g.coordinates.chromosome,
+          start: g.coordinates.start,
+          end: g.coordinates.end,
+          id: g.id,
+          name: g.name,
+        }
+      })
+      setOptions(r)
+      setGeneList(g)
+    } else if (genesSuggestion && genesSuggestion.length === 0) {
+      setOptions([])
+      setGeneList([])
+    }
+  }
+
+  // delay fetch
+  const debounceFn = useCallback(debounce(onSearchChange, 500), [])
+
+  const toggleDrawer = (open) => (event) => {
+    if (event.type === "keydown" && (event.key === "Tab" || event.key === "Shift")) {
+      return
+    }
+    setState(open)
+  }
+
+  // temp theme for toolbar color
+  const theme = createTheme({
+    palette: {
+      mode: "light",
+      // primary: {
+      //   main: "#nnn",
+      // },
+      secondary: {
+        main: "#nnn",
+      },
+    },
+    components: {
+      MuiAccordion: {
+        defaultProps: {
+          elevation: 0,
+        },
+      },
+    },
+  })
+
+  const drawerWidth: number = 350
+  let drawerHeight: number = window.screen.height
+  let drawerHeightTab: number = window.screen.height
+
+  // 1080
+  if (drawerHeight < 1200) {
```

Comment:
Ideally we shouldn't have manually set height ? @jpfisher72 any thoughts on how this can be done without manually setting height ?